src/file_management/core.clj

```diff
@@ -48,3 +56,26 @@
                    {:content-type "image/png"}
                    (s3/grant :all-users :read))
     (get-avatar-url credentials user-id))
+
+(defn gets-file-key
+  [file-key]
+  (str file-key "-" (java.util.UUID/randomUUID)))
```

Comment:
On your commit message you say that if the file-key exists it will be smashed. Don't this line prevent it?